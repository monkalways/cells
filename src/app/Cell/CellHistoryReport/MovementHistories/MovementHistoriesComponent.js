import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import {
  Grid,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
} from '@material-ui/core';
import { format } from 'date-fns';

const dateTimeFormat = 'yyyy/MM/dd HH:mm';

const CustomTableCell = withStyles((theme) => ({
  head: {
    backgroundColor: '#FFFFFF',
    fontSize: theme.typography.body1.fontSize,
    fontWeight: 800,
  },
  body: {
    fontSize: theme.typography.body1.fontSize,
  },
}))(TableCell);

const CustomTableRow = withStyles(() => ({
  root: {
    '&:nth-of-type(odd)': {
      backgroundColor: '#BEBFBF',
    },
    '&:nth-of-type(even)': {
      backgroundColor: '#FFFFFF',
    },
  },
}))(TableRow);

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  movementHistories: PropTypes.arrayOf(PropTypes.shape({
    time: PropTypes.instanceOf(Date).isRequired,
    detaineeName: PropTypes.string.isRequired,
    sourceCellName: PropTypes.string,
    destinationCellName: PropTypes.string.isRequired,
  })).isRequired,
};

const MovementHistoriesComponent = ({ classes, movementHistories }) => (
  <Grid item className={classes.container} xs={12}>
    <Table className={classes.table}>
      <TableHead>
        <TableRow>
          <CustomTableCell>Time</CustomTableCell>
          <CustomTableCell>Detainee</CustomTableCell>
          <CustomTableCell>Source Cell</CustomTableCell>
          <CustomTableCell>Destination Cell</CustomTableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {movementHistories.map((row) => (
          <CustomTableRow className={classes.row} key={row.time}>
            <CustomTableCell component="th" scope="row">
              {format(row.time, dateTimeFormat)}
            </CustomTableCell>
            <CustomTableCell>{row.detaineeName}</CustomTableCell>
            <CustomTableCell>{row.sourceCellName}</CustomTableCell>
            <CustomTableCell>{row.destinationCellName}</CustomTableCell>
          </CustomTableRow>
        ))}
      </TableBody>
    </Table>
  </Grid>
);

MovementHistoriesComponent.propTypes = propTypes;

export default withStyles(() => ({
  container: {
    width: '100%',
    overflowX: 'auto',
  },
}))(MovementHistoriesComponent);
