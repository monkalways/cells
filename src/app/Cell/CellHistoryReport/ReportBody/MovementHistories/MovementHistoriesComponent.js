import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import {
  Grid,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Typography,
} from '@material-ui/core';
import { format } from 'date-fns';

const dateTimeFormat = 'yyyy/MM/dd HH:mm';

const CustomTableCell = withStyles((theme) => ({
  head: {
    backgroundColor: '#FFFFFF',
    fontSize: theme.typography.body1.fontSize,
    fontWeight: 800,
  },
  body: {
    fontSize: theme.typography.body1.fontSize,
  },
}))(TableCell);

const CustomTableRow = withStyles(() => ({
  root: {
    '&:nth-of-type(odd)': {
      backgroundColor: '#BEBFBF',
    },
    '&:nth-of-type(even)': {
      backgroundColor: '#FFFFFF',
    },
  },
}))(TableRow);

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  movementHistories: PropTypes.arrayOf(PropTypes.shape({
    time: PropTypes.instanceOf(Date).isRequired,
    detaineeName: PropTypes.string.isRequired,
    sourceCellName: PropTypes.string,
    destinationCellName: PropTypes.string,
  })),
};

const defaultProps = {
  movementHistories: [],
};

export const MovementHistoriesComponent = ({ classes, movementHistories }) => (
  <Grid item className={classes.container} xs={12}>
    {movementHistories && movementHistories.length > 0 ? (
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <CustomTableCell>Time</CustomTableCell>
            <CustomTableCell>Detainee</CustomTableCell>
            <CustomTableCell>Source Cell</CustomTableCell>
            <CustomTableCell>Destination Cell</CustomTableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {movementHistories.map((row) => (
            <CustomTableRow className={classes.row} key={row.time}>
              <CustomTableCell component="th" scope="row" padding="dense">
                {format(row.time, dateTimeFormat)}
              </CustomTableCell>
              <CustomTableCell padding="dense">
                {row.detaineeName}
              </CustomTableCell>
              <CustomTableCell padding="dense">
                {row.sourceCellName}
              </CustomTableCell>
              <CustomTableCell padding="dense">
                {row.destinationCellName}
              </CustomTableCell>
            </CustomTableRow>
          ))}
        </TableBody>
      </Table>
    ) : (
      <Typography variant="body1" className={classes.text}>
        No detainee movements in given period.
      </Typography>
    )}
  </Grid>
);

MovementHistoriesComponent.propTypes = propTypes;
MovementHistoriesComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  container: {
    width: '100%',
    overflowX: 'auto',
  },
  text: {
    marginLeft: theme.spacing.unit,
  },
}))(MovementHistoriesComponent);
