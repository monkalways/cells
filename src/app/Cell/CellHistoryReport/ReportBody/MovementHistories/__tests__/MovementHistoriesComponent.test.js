import React from 'react';
import { shallow } from 'enzyme';
import { Table, Typography } from '@material-ui/core';
import { MovementHistoriesComponent } from '../MovementHistoriesComponent';

describe('MovementHistoriesComponent', () => {
  let movementHistories;

  beforeEach(() => {
    movementHistories = [
      {
        time: new Date(Date.UTC(2019, 2, 11, 0)),
        detaineeName: 'Test User 2',
        sourceCellName: 'c1',
        detinationCellName: 'c2',
      },
    ];
  });

  const setup = () => {
    const classes = {};
    return shallow(<MovementHistoriesComponent
      classes={classes}
      movementHistories={movementHistories}
    />);
  };

  it('should render movement histories if movementHistories is not empty', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Table)).toExist();
  });

  it('should render error message if movementHistories is empty', () => {
    movementHistories = null;

    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper
      .find(Typography)
      .render()
      .text()).toEqual('No detainee movements in given period.');
  });
});
