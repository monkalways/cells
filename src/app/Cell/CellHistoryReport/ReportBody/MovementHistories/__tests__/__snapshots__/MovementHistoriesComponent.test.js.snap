// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovementHistoriesComponent should render error message if movementHistories is empty 1`] = `
<WithStyles(Grid)
  item={true}
  xs={12}
>
  <WithStyles(Typography)
    variant="body1"
  >
    No detainee movements in given period.
  </WithStyles(Typography)>
</WithStyles(Grid)>
`;

exports[`MovementHistoriesComponent should render movement histories if movementHistories is not empty 1`] = `
<WithStyles(Grid)
  item={true}
  xs={12}
>
  <WithStyles(Table)>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(WithStyles(TableCell))>
          Time
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))>
          Detainee
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))>
          Source Cell
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))>
          Destination Cell
        </WithStyles(WithStyles(TableCell))>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(WithStyles(TableRow))
        key="Sun Mar 10 2019 18:00:00 GMT-0600 (Mountain Daylight Time)"
      >
        <WithStyles(WithStyles(TableCell))
          component="th"
          padding="dense"
          scope="row"
        >
          2019/03/10 18:00
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))
          padding="dense"
        >
          Test User 2
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))
          padding="dense"
        >
          c1
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))
          padding="dense"
        />
      </WithStyles(WithStyles(TableRow))>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</WithStyles(Grid)>
`;

exports[`ReportBodyComponent should render error message if movementHistories is empty 1`] = `
<WithStyles(Grid)
  item={true}
  xs={12}
>
  <WithStyles(Typography)
    variant="body1"
  >
    No detainee movements in given period.
  </WithStyles(Typography)>
</WithStyles(Grid)>
`;

exports[`ReportBodyComponent should render movement histories if movementHistories is not empty 1`] = `
<WithStyles(Grid)
  item={true}
  xs={12}
>
  <WithStyles(Table)>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(WithStyles(TableCell))>
          Time
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))>
          Detainee
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))>
          Source Cell
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))>
          Destination Cell
        </WithStyles(WithStyles(TableCell))>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(WithStyles(TableRow))
        key="Sun Mar 10 2019 18:00:00 GMT-0600 (Mountain Daylight Time)"
      >
        <WithStyles(WithStyles(TableCell))
          component="th"
          padding="dense"
          scope="row"
        >
          2019/03/10 18:00
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))
          padding="dense"
        >
          Test User 2
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))
          padding="dense"
        >
          c1
        </WithStyles(WithStyles(TableCell))>
        <WithStyles(WithStyles(TableCell))
          padding="dense"
        />
      </WithStyles(WithStyles(TableRow))>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</WithStyles(Grid)>
`;
